tsfile/src/main/java/org/apache/iotdb/tsfile/write/TsFileWriter.java

```diff
@@ -194,6 +187,35 @@ private boolean checkIsTimeSeriesExist(TSRecord record) throws WriteProcessExcep
     return true;
   }
 
+  /**
+   * Confirm whether the row batch is legal. If legal, add it into this RecordWriter.
+   *
+   * @param rowBatch - a row batch responding multiple columns
+   * @return - whether the row batch has been added into RecordWriter legally
+   * @throws WriteProcessException exception
+   */
+  private boolean checkIsTimeSeriesExist(RowBatch rowBatch) throws WriteProcessException {
+    IChunkGroupWriter groupWriter;
+    if (!groupWriters.containsKey(rowBatch.deviceId)) {
+      groupWriter = new ChunkGroupWriterImpl(rowBatch.deviceId);
+      groupWriters.put(rowBatch.deviceId, groupWriter);
+    } else {
+      groupWriter = groupWriters.get(rowBatch.deviceId);
+    }
+
+    // add all SeriesWriter of measurements in this RowBatch to this ChunkGroupWriter
+    Map<String, MeasurementSchema> schemaDescriptorMap = schema.getMeasurementSchemaMap();
+    for (MeasurementSchema measurement : rowBatch.measurements) {
+      String measurementId = measurement.getMeasurementId();
+      if (schemaDescriptorMap.containsKey(measurementId)) {
+        groupWriter.addSeriesWriter(schemaDescriptorMap.get(measurementId), pageSize);
+      } else {
+        throw new NoMeasurementException("input measurement is invalid: " + measurementId);
+      }
+    }
+    return true;
```

Comment:
You do not have to return this since it will never be "false", just use the exception.
You said "If legal, add it into this RecordWriter.", but I did not see the corresponding code.

Reply:
@jack870131  You can change this method to return void and remove the "return true;".

Also, remove the javadoc "If legal, add it into this RecordWriter."

Another: rename the IChunkGroupWriter.addSeriesWriter to tryToAddSeriesWriter

Reply:
Done.